### PR TITLE
Add support for extending the cluster with addons

### PR DESCRIPTION
**Pull request by wandera-petr-cerny**

### CODE REVIEW

File: aws/cluster/cluster-spec.tf [hunk 1]
```diff
@@ -287,3 +290,26 @@ locals {
...
+
+data "aws_iam_policy_document" "master-additional" {
+  source_json = var.master-additional-policies
+
+  statement {
+    actions = [
+      "s3:GetObject",
+    ]
+    resources = [
+      "arn:aws:s3:::${var.kops-state-bucket}/terraform-addons/${var.cluster-name}/*"
```
Comment (gerricchaplin) [inline aws/cluster/cluster-spec.tf:302]: Any reason to use a top-level `terraform-addons/${var.cluster-name}/` as opposed to `${var.cluster-name}/terraform-addons/` which would mean you probably don't need this policy althogether?
`s3://${var.kops-state-bucket}/${var.cluster-name}/terraform-addons/` in other words? 
Its quite tightly coupled?

File: aws/cluster/variables.tf [hunk 2]
```diff
@@ -96,6 +96,17 @@ variable "kops-state-bucket" {
   description = "Name of the bucket in which kops stores its state (must be created prior to cluster turnup)"
 }
 
+variable "kops-addons" {
```
Comment (coufalja) [inline aws/cluster/variables.tf:99]: Note that there are 2 types of KOPS addons, a static (manifest) ones and built-in managed ones. https://kops.sigs.k8s.io/addons/ the variable should either support both or be future proof. e.g. what about `kops-addon-manifests` or different name?

**Review by coufalja — CHANGES_REQUESTED** [review]

File: aws/cluster/cluster-spec.tf (hunk 1 above)
Comment (wandera-petr-cerny) [reply aws/cluster/cluster-spec.tf:302]: I had your proposed version implemented originally, but kops refuses to destroy cluster if there are files on that path that it didn’t create, which is why I have changed the directory structure.

File: aws/cluster/variables.tf (hunk 2 above)
Comment (wandera-petr-cerny) [reply aws/cluster/variables.tf:99]: Good point, I was checking only docu for kops version we use now (1.14.1). In order to be future-proof and match the official docu, I would go for `kops-static-addons` and add a link to the currently latest docu. https://github.com/kubernetes/kops/blob/1ae09e86a5e48e16cb885b8d001023cd1894d59e/docs/addons.md

**Review by coufalja — APPROVED** [review]

**Review by gerricchaplin — APPROVED** [review]

**Pull request merged**